sstables/mx/writer.cc

```diff
@@ -840,6 +842,30 @@ void writer::add_pi_block() {
     _pi_write_m.offsets_serialized_size += offsets.size();
 
     _pi_write_m.blocks.emplace_back(std::move(block));
+
+    // auto-scale?
+    if (_pi_write_m.auto_scale_threshold // enabled
+            && _pi_write_m.blocks.size() >= _pi_write_m.auto_scale_threshold // crossed threshold
+            && !(_pi_write_m.blocks.size() % 2)) { // has even number of blocks
+        utils::chunked_vector<pi_block> new_blocks;
+        new_blocks.reserve(_pi_write_m.blocks.size() / 2);
+        blocks = measuring_output_stream();
+        offsets = measuring_output_stream();
+        auto it = _pi_write_m.blocks.cbegin();
+        while (it != _pi_write_m.blocks.cend()) {
+            block = *it++;
+            auto block2 = *it++;
+            block.last = std::move(block2.last);
+            block.width += block2.width;
+            block.open_marker = block2.open_marker;
+            write_pi_block(blocks, offsets, block, blocks.size());
+            new_blocks.emplace_back(std::move(block));
```

Comment:
and yield, and deallocate the previous two blocks.

Reply:
yes (if we keep this patch)